### PR TITLE
Assaf's doc improvements 

### PR DESCRIPTION
**Pull request by lenak25**

### CODE REVIEW

**Comment by lenak25** [comment]
@AvishaiW , please review :)

File: docs/ARCHITECTURE.md [hunk 1]
```diff
@@ -319,7 +319,19 @@ There are more things in [Worker](https://github.com/enigmampc/enigma-p2p/tree/m
...
-* TBD
+
+```bash
+sudo npm install -g truffle
+git clone git@github.com:enigmampc/enigma-p2p.git
+cd enigma-p2p
+npm install
+npm run test
+```
+
```
Comment (AvishaiW) [inline docs/ARCHITECTURE.md]: maybe add a `Troubleshooting` title or something like that? It's not critical...

Comment (lenak25) [reply docs/ARCHITECTURE.md]: Added, thanks!

**Comment by lenak25** [comment]
@AvishaiW , your comment was integrated, please review

File: docs/ARCHITECTURE.md [hunk 2]
```diff
@@ -328,8 +328,7 @@ npm install
 npm run test
 ```
 
-Currently v8.0.0 <= Node.JS <= v11.15.0 is required because web3 v1.0.0-beta.54 requires it. In the future this restriction will be lifted.
```
Comment (AvishaiW) [inline docs/ARCHITECTURE.md]: why did you remove this?

File: docs/ARCHITECTURE.md (hunk 1 above)
Comment (AvishaiW) [reply docs/ARCHITECTURE.md]: :)

File: docs/ARCHITECTURE.md (hunk 2 above)
Comment (lenak25) [reply docs/ARCHITECTURE.md]: Because of [this PR](https://github.com/enigmampc/enigma-p2p/pull/220)
